runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeOfferSlot.java

```diff
@@ -72,123 +89,189 @@
-		JPanel offerStatePanel = new JPanel();
-		offerStatePanel.setLayout(new BoxLayout(offerStatePanel, BoxLayout.Y_AXIS));
-		offerStatePanel.add(offerState);
-		offerStatePanel.add(progressBar);
-		infoCard.add(offerStatePanel);
-
-		// Card for when the slot is empty
-		JPanel emptySlotCard = new JPanel();
-		add(emptySlotCard, EMPTY_CARD);
-		// Counteract the height lost to the text at the top of the TitledBorder
-		int itemNameBorderHeight = itemName.getBorderInsets(this).top;
-		emptySlotCard.setBorder(BorderFactory.createEmptyBorder(0, 0, (itemNameBorderHeight - 1) / 2, 0));
-		// Center the "Empty" label horizontally
-		emptySlotCard.setLayout( new BoxLayout(emptySlotCard, BoxLayout.X_AXIS));
-		emptySlotCard.add(Box.createHorizontalGlue());
-		emptySlotCard.add(new JLabel(getNameForState(EMPTY)), BorderLayout.CENTER);
-		emptySlotCard.add(Box.createHorizontalGlue());
-
-		cardLayout.show(this, EMPTY_CARD);
+
+		setLayout(new BorderLayout());
+		container.setLayout(cardLayout);
+
+		try
+		{
+			BufferedImage rightImg;
+			BufferedImage leftImg;
+			synchronized (ImageIO.class)
+			{
+				rightImg = ImageIO.read(GrandExchangePlugin.class.getResourceAsStream("arrow_right.png"));
+				leftImg = ImageIO.read(GrandExchangePlugin.class.getResourceAsStream("arrow_left.png"));
+			}
+			rightArrowIcon = new ImageIcon(rightImg);
+			leftArrowIcon = new ImageIcon(leftImg);
+		}
+		catch (IOException e)
+		{
+			log.warn("Failed to read icon", e);
+		}
+
+		JPanel faceCard = new JPanel();
+		faceCard.setBackground(ColorScheme.MEDIUM_GRAY_COLOR);
+		faceCard.setLayout(new BorderLayout());
+		faceCard.addMouseListener(new MouseAdapter()
+		{
+			@Override
+			public void mousePressed(MouseEvent mouseEvent)
+			{
+				super.mousePressed(mouseEvent);
+				switchPanel();
+			}
+		});
+
+		itemIcon.setVerticalAlignment(JLabel.CENTER);
+		itemIcon.setHorizontalAlignment(JLabel.CENTER);
+		itemIcon.setPreferredSize(new Dimension(45, 45));
+
+		itemName.setForeground(Color.WHITE);
+		itemName.setVerticalAlignment(JLabel.BOTTOM);
+		itemName.setFont(FontManager.getRunescapeSmallFont());
+
+		offerInfo.setForeground(ColorScheme.LIGHT_GRAY_COLOR);
+		offerInfo.setVerticalAlignment(JLabel.TOP);
+		offerInfo.setFont(FontManager.getRunescapeSmallFont());
+
+		switchFaceViewIcon.setIcon(rightArrowIcon);
+		switchFaceViewIcon.setVerticalAlignment(JLabel.CENTER);
+		switchFaceViewIcon.setHorizontalAlignment(JLabel.CENTER);
+		switchFaceViewIcon.setPreferredSize(new Dimension(30, 45));
+
+
+		JPanel offerFaceDetails = new JPanel();
+		offerFaceDetails.setOpaque(false);
+		offerFaceDetails.setLayout(new GridLayout(2, 1, 0, 2));
+
+		offerFaceDetails.add(itemName);
+		offerFaceDetails.add(offerInfo);
+
+		faceCard.add(offerFaceDetails, BorderLayout.CENTER);
+		faceCard.add(itemIcon, BorderLayout.WEST);
+		faceCard.add(switchFaceViewIcon, BorderLayout.EAST);
+
+
+		/* -------------------------------------------------------------------------------------- */
```

Comment:
So many new lines :O

Reply:
My bad, was supposed to clean up afterwards, that was just a little visual seperator to keep me organized.